src/platform/channel/unix.rs

```diff
@@ -99,15 +99,18 @@ impl UnixChannel {
         }
         let mut buf = [0u8; 4];
         let mut total_bytes = 0;
-        let some_ready = if wait {
+        let ready_ids_result = if wait {
             let mut timeout = TimeVal::zero();
-            let num_of_ready_fds =
-                select(None, Some(&mut read_set), None, None, Some(&mut timeout))?;
-            num_of_ready_fds != 0
+            select(None, Some(&mut read_set), None, None, Some(&mut timeout))
         } else {
-            let num_of_ready_fds = select(None, Some(&mut read_set), None, None, None)?;
-            num_of_ready_fds != 0
+            select(None, Some(&mut read_set), None, None, None)
         };
+        let some_ready = match ready_ids_result {
+            Ok(num) => num != 0,
+            Err(nix::Error::Sys(_)) => true,
+            Err(e) => Err(e)?,
+        };
+
```

Comment:
Instead of setting `some_ready` and continuing the execution to `unistd::read` if an nix error is received, the logic should be following: call `select` and check the resulting value. The value should be handled the same as for `unistd::read` a couple of lines below, which is: on `Ok`, return the number of ready file descriptors, if the error received is `EINTR` call select again with the same arguments, otherwise fail.